File: QueueUsingArray.java
class Queue
{
    int size = 4 , front = -1, rear = -1;
    int[] queue = new int[size];

    void Insert(int data)
    {
        if(rear >= size-1)
        {
            System.out.println("Queue OverFlow"+"\n"+"Cannot Insert Elements");
        }
        else
        {
            rear++;
            queue[rear] = data;

            if(front == -1)
            {
                front++;
            }
        }
    }

    void QDel()
    {
        if(front == -1)
        {
            System.out.println("Queue UnderFlow"+"\n"+"Cannot Delete Elements");
        }
        else
        {
            int temp = queue[front];
            queue[front] = 0;
             System.out.println("Deleted Data: "+temp+"\n");
            
            if(front == rear)
            {
                front = rear = -1;
                System.out.println("Front: "+front+"\t"+"Rear: "+rear+"\n");
            }
            front++;
        }
    }

    void Display()
    {
        while(rear > -1)
        {
            System.out.println("Data: "+queue[rear]+"\t"+"Front: "+front+"\t"+"Rear: "+rear+"\n");
            rear--;
        }
    }
}

public class QueueUsingArray
{

    public static void main(String args[])
    {
        Queue q = new Queue();

        q.Insert(10);
        q.Insert(20);
        q.Insert(30);
        q.Insert(40);

        //q.QDel();
        //q.QDel();
        //q.QDel();
        //q.QDel();

        q.Display();
    }
}

File: Queue Using Array.c
#include<stdio.h>
#define MAX 4

void insert(int *queue, int *front, int *rear, int data)
{
	if(*rear >= MAX-1)
	{
		printf("Queue OverFlow\n Cannot Insert Elements\n");
	}
	else
	{
		(*rear)++;
		queue[*rear] = data;
		
		if(*front == -1)
		{
			(*front)++;
		}
	}
}

int qdelete(int *queue, int *front, int *rear)
{
	if(*front == -1)
	{
		printf("Queue is UnderFlow\n Cannot Delete Elements");
	}
	else
	{
		int data = queue[*front];
		printf("\nDeleted Data: %i\n",data);
		queue[*front] = NULL;
		
		if(*front == *rear)
		{
			*front = *rear = -1;
			printf("Front: %i\t Rear: %i\n",*front,*rear);
		}
		(*front)++;
		return data;
	}
}

void display(int *queue, int *front, int *rear)
{
	while(*rear > -1)
	{
		printf("Data: %i\t Front: %i\t Rear: %i\n",queue[*rear], *front, *rear);
		(*rear)--;
	}
}

void main()
{
	int queue[MAX], front, rear;
	front = -1;
	rear = -1;
	
	insert(queue, &front, &rear, 10);
	insert(queue, &front, &rear, 20);
	insert(queue, &front, &rear, 30);
	insert(queue, &front, &rear, 40);
	
//	qdelete(queue, &front, &rear);
//	qdelete(queue, &front, &rear);
//	qdelete(queue, &front, &rear);
//	qdelete(queue, &front, &rear);
	
	display(queue, &front, &rear);
}
